nyxx/lib/src/utils/EntityUtility.dart

```diff
@@ -0,0 +1,139 @@
+part of nyxx;
+
+/// Entity Utility is a Utility that lets you create Entities from outside of the nyxx project. Typically used in nyxx.* projects.
+/// An example getting a user is below:
+/// ```dart
+/// void main() {
+///     var bot = Nyxx("TOKEN");
+///     Map<String, dynamic> json = /* JSON from the API */
+///     User user = EntityUtility.createUser(bot, json);
+/// }
+/// ```
+class EntityUtility {
+  /// Creates a User object, can be used for other classes where you have correct JSON data from the API.
+  /// ```dart
+  /// void main() {
+  ///     var bot = Nyxx("TOKEN");
+  ///     Map<String, dynamic> json = /* JSON from the API */
+  ///     User user = EntityUtility.createUser(bot, json);
+  /// }
+  /// ```
+  User createUser(
+    Nyxx client,
+    Map<String, dynamic> json,
+  ) =>
+      User._new(
+        client,
+        json,
+      );
+
+  /// Creates a Guild object, can be used for other classes where you have correct JSON data from the API.
+  /// ```dart
+  /// void main() {
+  ///     var bot = Nyxx("TOKEN");
+  ///     Map<String, dynamic> json = /* JSON from the API */
+  ///     Guild guild = EntityUtility.createGuild(bot, json);
+  /// }
+  /// ```
+  Guild createGuild(
+    Nyxx client,
+    Map<String, dynamic> json,
+  ) =>
+      Guild._new(
+        client,
+        json,
+      );
+
+  /// Creates a Role object, can be used for other classes where you have correct JSON data from the API.
+  /// ```dart
+  /// void main() {
+  ///     var bot = Nyxx("TOKEN");
+  ///     Map<String, dynamic> json = /* JSON from the API */
+  ///     Role role = EntityUtility.createRole(bot, json, Snowflake("81384788765712384"));
+  /// }
+  /// ```
+  Role createRole(
+    Nyxx client,
+    Snowflake guildId,
+    Map<String, dynamic> json,
+  ) =>
+      Role._new(
+        client,
+        json,
+        guildId,
+      );
+
+  /// Creates a CategoryGuildChannel object, can be used for other classes where you have correct JSON data from the API.
+  /// ```dart
+  /// void main() {
+  ///     var bot = Nyxx("TOKEN");
+  ///     Map<String, dynamic> json = /* JSON from the API */
+  ///     CategoryGuildChannel category = EntityUtility.createCategoryGuildChannel(bot, json, Snowflake("81384788765712384"));
```

Comment:
example has arguments in wrong order

Reply:
Checked all examples and updated to match EntityUtility & CacheUtility param order